### PR TITLE
fixes: 1385 fail test if not valid success message

### PR DESCRIPTION
**Pull request by villesau**
This is naive solution for the https://github.com/flowtype/flow-typed/issues/1385 . Flow returns errCode 0 if it only detects warnings so it can't be used unless someone makes changes to Flow it self.

Failing on all the warnings might not be bad idea either as type definitions should follow as good practices as possible imo.

Testing the type definitions would benefit a lot of tests too. This bug could perhaps have been caught by them.

Fixing this might reveal quite a lot of broken type definitions.

### CODE REVIEW

**Comment by aaronjensen** [comment]
It looks like include_warnings isn't valid in older flow versions. You'll probably need to write a different flowconfig based on the version

**Comment by villesau** [comment]
That's actually the tricky part! seems that the type definition tests are run in parallel and they all uses the same flowconfig. Flow doesn't allow defining the .flowconfig path separately so i can't define it differently for each version.

The range can overlap between 0.52.x and 0.53.0 when the include_warnings was introduced so using batch version range is not feasible.

**Comment by villesau** [comment]
Now the runner writes .flowConfig for each version separately, but that means that the tests are run in series which most likely makes them way too slow.

**Comment by villesau** [comment]
Now it only creates new .flowconfig when needed and run tests in batches again.

**Comment by GAntoine** [comment]
I was working on another solution, but stopped when notices how similar it was to this one. I'll take a look at it this evening.

File: cli/src/commands/runTests.js [hunk 1]
```diff
@@ -354,61 +415,28 @@ async function runTestGroup(
...
+    );
+    const lowerVersions = flowVersionsToRun.filter(flowVer =>
+      semver.lt(flowVer, "0.53.0")
+    );
 
-            child.on("close", errCode => {
-              res({ stdErrOut, errCode, execError: null });
-            });
-          });
+    await writeFlowConfig(testDirPath, testGroup.libDefPath, true);
```
Comment (GAntoine) [inline cli/src/commands/runTests.js]: Could you switch there around to run the lower versions first?

**Comment by villesau** [comment]
I also changed the version hinting check to work in the same way as the actual tests.

**Pull request merged**